internal/transport/http2_server.go

```diff
@@ -371,32 +371,43 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 			timeoutSet = true
 			var err error
 			if timeout, err = decodeTimeout(hf.Value); err != nil {
-				headerError = true
+				headerError = err
 			}
 		default:
 			if isReservedHeader(hf.Name) && !isWhitelistedHeader(hf.Name) {
 				break
 			}
 			v, err := decodeMetadataHeader(hf.Name, hf.Value)
 			if err != nil {
-				headerError = true
+				headerError = err
 				logger.Warningf("Failed to decode metadata header (%q, %q): %v", hf.Name, hf.Value, err)
 				break
 			}
 			mdata[hf.Name] = append(mdata[hf.Name], v)
 		}
 	}
 
-	if !isGRPC || headerError {
-		t.controlBuf.put(&cleanupStream{
+	if !isGRPC {
+		t.controlBuf.put(&headerFrame{
 			streamID: streamID,
-			rst:      true,
-			rstCode:  http2.ErrCodeProtocol,
-			onWrite:  func() {},
+			hf: []hpack.HeaderField{
+				{Name: ":status", Value: "415"},
+			},
+			endStream: true,
 		})
 		return false
 	}
 
+	if headerError != nil {
+		if err := t.WriteStatus(
+			s,
+			status.Newf(codes.Internal, "transport: http2Server.operateHeaders failed parse headers: %v", headerError),
+		); err != nil && logger.V(logLevel) {
+			logger.Errorf("transport: http2Server.operateHeaders failed to write status %v", err)
```

Comment:
Make this `Info` since it's behind a `V`.

Reply:
Done